src/StateMachineContext.tsx

```diff
@@ -15,9 +15,9 @@ export const StateMachineProvider: React.FC<PropsChildren> = ({ children }) => {
   const [state, setState] = React.useState(storeFactory.state);
 
   React.useEffect(() => {
-    if (storeFactory.options.persist === PERSIST_BEFORE_UNLOAD) {
+    if (storeFactory.options.persist === PersistOption.BeforeUnload) {
       window.onbeforeunload = () => storeFactory.saveStore();
-      storeFactory.options.storageType.removeItem(storeFactory.options.name);
+      storeFactory.options.storageType?.removeItem(storeFactory.options.name!);
```

Comment:
could we avoid the optional chain operator? it increases the package size during transpile.